src/utilities/__tests__/resolveSchemaCoordinate-test.ts

```diff
@@ -0,0 +1,239 @@
+    ).to.deep.equal(undefined);
+  });
+
+  it('resolves a Enum Value', () => {
+    const type = schema.getType('SearchFilter') as GraphQLEnumType;
+    const enumValue = type.getValue('OPEN_NOW');
+    expect(
+      resolveSchemaCoordinate(schema, 'SearchFilter.OPEN_NOW'),
+    ).to.deep.equal({
+      kind: 'EnumValue',
+      type,
+      enumValue,
+    });
+
+    expect(
+      resolveSchemaCoordinate(schema, 'SearchFilter.UNKNOWN'),
+    ).to.deep.equal(undefined);
+  });
+
+  it('resolves a Field Argument', () => {
+    const type = schema.getType('Query') as GraphQLObjectType;
+    const field = type.getFields().searchBusiness;
+    const fieldArgument = field.args.find((arg) => arg.name === 'criteria');
+    expect(
+      resolveSchemaCoordinate(schema, 'Query.searchBusiness(criteria:)'),
+    ).to.deep.equal({
+      kind: 'FieldArgument',
+      type,
+      field,
+      fieldArgument,
+    });
+
+    expect(
+      resolveSchemaCoordinate(schema, 'Business.name(unknown:)'),
+    ).to.deep.equal(undefined);
+
+    expect(() =>
+      resolveSchemaCoordinate(schema, 'Unknown.field(arg:)'),
+    ).to.throw('Expected "Unknown" to be defined as a type in the schema.');
+
+    expect(() =>
+      resolveSchemaCoordinate(schema, 'Business.unknown(arg:)'),
+    ).to.throw(
+      'Expected "unknown" to exist as a field of type "Business" in the schema.',
+    );
+
+    expect(() =>
+      resolveSchemaCoordinate(schema, 'SearchCriteria.name(arg:)'),
+    ).to.throw(
+      'Expected "SearchCriteria" to be an object type or interface type.',
+    );
+  });
+
+  it('resolves a Directive', () => {
+    expect(resolveSchemaCoordinate(schema, '@private')).to.deep.equal({
+      kind: 'Directive',
+      directive: schema.getDirective('private'),
+    });
+
+    expect(resolveSchemaCoordinate(schema, '@deprecated')).to.deep.equal({
+      kind: 'Directive',
+      directive: schema.getDirective('deprecated'),
+    });
+
+    expect(resolveSchemaCoordinate(schema, '@unknown')).to.deep.equal(
+      undefined,
+    );
+
+    expect(resolveSchemaCoordinate(schema, '@Business')).to.deep.equal(
+      undefined,
+    );
+  });
+
+  it('resolves a Directive Argument', () => {
+    const directive = schema.getDirective('private') as GraphQLDirective;
+    const directiveArgument = directive.args.find(
+      (arg) => arg.name === 'scope',
+    );
+    expect(resolveSchemaCoordinate(schema, '@private(scope:)')).to.deep.equal({
+      kind: 'DirectiveArgument',
+      directive,
+      directiveArgument,
+    });
+
+    expect(resolveSchemaCoordinate(schema, '@private(unknown:)')).to.deep.equal(
+      undefined,
+    );
+
+    expect(() => resolveSchemaCoordinate(schema, '@unknown(arg:)')).to.throw(
+      'Expected "unknown" to be defined as a directive in the schema.',
+    );
+  });
+
+  it('resolves a meta-field', () => {
+    const type = schema.getType('Business') as GraphQLObjectType;
+    const field = schema.getField(type, '__typename');
```

Comment:
The implementation of `schema.getField()` may change in the future to return undefined for metafields. Let's ensure the test throws in that case:

```suggestion
    const field = schema.getField(type, '__typename');
    assert.ok(field);
```

(Not sure if this is the correct code; but I've deliberately used `assert()` rather than `expect()` here to indicate it's an assumption of the test rather than something actually being tested _for_.)

Hopefully this one assertion means we won't need to add more in the other tests, but feel free to do so if you're so inclined.